src/protocols/secure_channel/StatusReport.h

```diff
@@ -0,0 +1,103 @@
+/**
+ *    @file
+ *      Defines the StatusReport class and General Codes for StatusReport messages.
+ */
+
+#pragma once
+
+#include <support/BufferWriter.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace Protocols {
+namespace SecureChannel {
+
+/**
+ *  Encapsulates the data included in a StatusReport message, and provides methods for writing to and reading from PacketBuffers
+ *  that contain StatusReport messages.
+ */
+class DLL_EXPORT StatusReport
+{
+public:
+    /**
+     *  Construct a StatusReport with zero-d out fields (for use before calling \c Parse() ).
+     */
+    StatusReport();
+
+    /**
+     *  Construct a StatusReport with no additional ProtocolData.
+     *
+     *  @param generalCode Required, one of the \c GeneralStatusCode values listed in \c secure_channel/Constants.h
+     *  @param protocolId Must specify a ProtocolId which consists of Vendor Id (upper 16 bits) and ProtocolId (lower 16 bits)
+     *  @param protocolCode A code defined by the specified protocol which provides more information about the status
+     */
+    StatusReport(uint16_t generalCode, uint32_t protocolId, uint16_t protocolCode);
+
+    //
+    /**
+     *  Construct a StatusReport with additional ProtocolData.
+     *
+     *  WARNING: The caller MUST ensure that the data referenced by protocolData not be freed before calling \c WriteToBuffer()
```

Comment:
```suggestion
     *  WARNING: The caller MUST ensure that the data referenced by protocolData will not be freed before calling \c WriteToBuffer()
```

Reply:
This comment has been removed since switching to use PacketBufferHandle instead of a pointer